cpp/oneapi/dal/table/backend/interop/host_homogen_table_adapter.cpp

```diff
@@ -0,0 +1,394 @@
+    const std::int64_t row_count = original_table_.get_row_count();
+    return info.row_begin_index < row_count && info.row_end_index <= row_count;
+}
+
+template <typename Data>
+bool host_homogen_table_adapter<Data>::check_column_index_in_range(const block_info& info) const {
+    const std::int64_t column_count = original_table_.get_column_count();
+    return info.single_column_requested && info.column_index < column_count;
+}
+
+template <typename Data>
+host_homogen_table_adapter<Data>::host_homogen_table_adapter(const homogen_table& table,
+                                                             status_t& stat)
+        // The following const_cast is safe only when this class is used for read-only
+        // operations. Use on write leads to undefined behaviour.
+        : base(daal::data_management::DictionaryIface::equal,
+               ptr_data_t{ const_cast<Data*>(table.get_data<Data>()), daal_object_owner(table) },
+               table.get_column_count(),
+               table.get_row_count(),
+               stat),
+          is_rowmajor_(table.get_data_layout() == data_layout::row_major) {
+    if (!stat.ok()) {
+        return;
+    }
+    else if (!table.has_data()) {
+        stat.add(daal::services::ErrorIncorrectParameter);
+        return;
+    }
+
+    original_table_ = table;
+
+    this->_memStatus = daal::data_management::NumericTableIface::userAllocated;
+    this->_layout = daal::data_management::NumericTableIface::aos;
+
+    auto& daal_dictionary = *this->getDictionarySharedPtr();
+    convert_feature_information_to_daal(original_table_.get_metadata(), daal_dictionary);
+}
+
+template class host_homogen_table_adapter<std::int32_t>;
+template class host_homogen_table_adapter<float>;
+template class host_homogen_table_adapter<double>;
+
+#define INSTANTIATE_READ_COLUMN_VALUES_IMPL(DataType, BlockType)                 \
+    template auto host_homogen_table_adapter<DataType>::read_column_values_impl( \
+        std::size_t,                                                             \
+        std::size_t,                                                             \
+        std::size_t,                                                             \
+        rw_mode_t,                                                               \
+        block_desc_t<BlockType>&)                                                \
+        ->status_t;
+
+#define INSTANTIATE_READ_ROWS_IMPL(DataType, BlockType)                                          \
+    template auto host_homogen_table_adapter<DataType>::read_rows_impl(std::size_t,              \
+                                                                       std::size_t,              \
+                                                                       rw_mode_t,                \
+                                                                       block_desc_t<BlockType>&) \
+        ->status_t;
```

Comment:
Do you really need these instantiations, aren't they instantiated automatically? These method explicitly called by non-template methods of `homogen_table_adapter` and should be instantiated along with class instantiation.

Maybe Windows requires this...

Reply:
Let's remove this code while we do not see any problems